packages/zwave-js/src/lib/serialapi/capability/GetSerialApiInitDataMessages.ts

```diff
@@ -215,3 +218,115 @@ export class GetSerialApiInitDataResponse extends Message {
 //                         is SUC:             true
 //                         chip type:          7
 //                         chip version:       0
+
+export interface GetLongRangeNodesRequestOptions extends MessageBaseOptions {
+	listStartOffset128: number;
+}
+
+@messageTypes(MessageType.Request, FunctionType.GetLongRangeNodes)
+@expectedResponse(FunctionType.GetLongRangeNodes)
+@priority(MessagePriority.Controller)
+export class GetLongRangeNodesRequest extends Message {
+	public constructor(
+		host: ZWaveHost,
+		options:
+			| MessageDeserializationOptions
+			| GetLongRangeNodesRequestOptions,
+	) {
+		super(host, options);
+
+		if (gotDeserializationOptions(options)) {
+			// BUGBUG: validate length at least
+			this.listStartOffset128 = this.payload[0];
+		} else {
+			this.listStartOffset128 = options.listStartOffset128;
+		}
+	}
+
+	public listStartOffset128: number;
+
+	public serialize(): Buffer {
+		this.payload = Buffer.allocUnsafe(
+			1,
+		);
+
+		this.payload[0] = this.listStartOffset128;
+		return super.serialize();
+	}
+}
+
+export interface GetLongRangeNodesResponseOptions extends MessageBaseOptions {
+	moreNodes: boolean;
+	listStartOffset128: number;
+	nodeIds: number[];
+}
+
+@messageTypes(MessageType.Response, FunctionType.GetLongRangeNodes)
+export class GetLongRangeNodesResponse extends Message {
+	public constructor(
+		host: ZWaveHost,
+		options:
+			| MessageDeserializationOptions
+			| GetLongRangeNodesResponseOptions,
+	) {
+		super(host, options);
+
+		if (gotDeserializationOptions(options)) {
+			// BUGBUG: validate length at least
+			this.moreNodes = this.payload[0] != 0;
+			this.listStartOffset128 = this.payload[1];
+			const listLength = this.payload[2];
+
+			const listStart = 3;
+			const listEnd = listStart + listLength;
+			// BUGGUG validate listEnd <= this.payload.length
+			if (listEnd <= this.payload.length) {
+				const nodeBitMask = this.payload.subarray(
+					listStart,
+					listEnd,
+				);
+				this.nodeIds = parseLongRangeNodeBitMask(
+					nodeBitMask,
+					this.listStartNode(),
+				);
+			} else {
+				this.nodeIds = [];
+			}
+		} else {
+			this.moreNodes = options.moreNodes;
+			this.listStartOffset128 = options.listStartOffset128;
+			this.nodeIds = options.nodeIds;
+		}
+	}
+
+	public moreNodes: boolean;
+	public listStartOffset128: number;
+	public nodeIds: readonly number[];
+
+	public serialize(): Buffer {
+		this.payload = Buffer.allocUnsafe(
+			3 + NUM_LR_NODEMASK_SEGMENT_BYTES,
+		);
+
+		this.payload[0] = this.moreNodes ? 1 : 0;
+		this.payload[1] = this.listStartOffset128;
+		this.payload[2] = NUM_LR_NODEMASK_SEGMENT_BYTES;
+
+		const nodeBitMask = encodeLongRangeNodeBitMask(
+			this.nodeIds,
+			this.listStartNode(),
+		);
+		nodeBitMask.copy(this.payload, 3);
+
+		return super.serialize();
+	}
+
+	private listStartNode(): number {
+		return 256 + NUM_LR_NODES_PER_SEGMENT * 8 * this.listStartOffset128;
+	}
+
```

Comment:
The equations given in the spec for this say `255 +`, but the given examples result in this being `256 +`. What's here is correct experimentally. Maybe the spec was using 1-based math and I missed it?

Reply:
I think 256 makes more sense. Not the first time that the specs are wrong.

Reply:
The `* 8` seems wrong though?